src/platform_impl/linux/kms/mod.rs

```diff
@@ -0,0 +1,166 @@
+use crate::dpi::{PhysicalPosition, PhysicalSize};
+
+pub mod event_loop;
+pub mod input;
+pub mod window;
+use crate::platform_impl;
+pub use drm::SystemError;
+use drm::{
+    control::{Device as ControlDevice, *},
+    Device,
+};
+pub use event_loop::EventLoop;
+pub use event_loop::EventLoopProxy;
+pub use event_loop::EventLoopWindowTarget;
+use parking_lot::Mutex;
+use std::os::unix;
+use std::os::unix::prelude::FromRawFd;
+use std::sync::Arc;
+pub use window::Window;
+
+pub static MODE: Mutex<Option<Mode>> = parking_lot::const_mutex(None);
+
+#[derive(Debug, Clone)]
+/// A simple wrapper for a device node.
+pub struct Card(pub(crate) Arc<i32>);
+
+/// Implementing `AsRawFd` is a prerequisite to implementing the traits found
+/// in this crate. Here, we are just calling `as_raw_fd()` on the inner File.
+impl unix::io::AsRawFd for Card {
+    fn as_raw_fd(&self) -> unix::io::RawFd {
+        *self.0
+    }
+}
```

Comment:
Modern versions of the `drm` crate use `AsFd`, maybe we should use that instead?